poly/src/domain/radix2/fft.rs

```diff
@@ -75,19 +75,27 @@ impl<F: FftField> Radix2EvaluationDomain<F> {
         let mut gap = xi.len() / 2;
         while gap > 0 {
             // each butterfly cluster uses 2*gap positions
-            let nchunks = xi.len() / (2 * gap);
-            ark_std::cfg_chunks_mut!(xi, 2 * gap).for_each(|cxi| {
+            let chunk_size = 2 * gap;
+            let nchunks = xi.len() / chunk_size;
+
+            let butterfly_fn = |(idx, (lo, hi)): (usize, (&mut T, &mut T))| {
+                let neg = *lo - *hi;
+                *lo += *hi;
+
+                *hi = neg;
+                *hi *= roots[nchunks * idx];
+            };
+
+            // If the gap is sufficiently big that parallelism helps,
+            // we parallelize computation of values in the gap.
+            // Notice that the core loops are the same in both cases
+            ark_std::cfg_chunks_mut!(xi, chunk_size).for_each(|cxi| {
                 let (lo, hi) = cxi.split_at_mut(gap);
-                ark_std::cfg_iter_mut!(lo, 1000) // threshold of 1000 was determined empirically
-                    .zip(hi)
-                    .enumerate()
-                    .for_each(|(idx, (lo, hi))| {
-                        let neg = *lo - *hi;
-                        *lo += *hi;
-
-                        *hi = neg;
-                        *hi *= roots[nchunks * idx];
-                    });
+                if gap > MIN_CHUNK_SIZE_FOR_PARALLELIZATION / 2 {
```

Comment:
```suggestion
                if chunk_size > MIN_CHUNK_SIZE_FOR_PARALLELIZATION {
```

Reply:
Oh I just saw that the latest change comment reverted this change; is there a reason why?

Reply:
yeah, really oddly the 2^25 degree FFT was saying it was 1% slower with the `chunk_size > ` form.

 I didn't compile in god bolt to check if `chunk_size` is really taking an extra register, or if its just system noise. In the case that it is taking an extra register, we could just un-nest the if statement.

Reply:
In that case but we might want to just remove chunk_size, right?

Reply:
I prefer the constant being called `chunk_size` though. I'll add a comment about the if statement stating the condition in terms of chunk size?

Reply:
Alright, sounds good then, this PR is good to go =)